src/libp2p_transport_tcp.erl

```diff
@@ -343,9 +343,14 @@ listen_on(Addr, TID) ->
                                               DefaultListenOpts),
             % Dialyzer severely dislikes ranch_tcp:listen so we
             % emulate it's behavior here
+            Port = reuseport0(TID, Type, IP, Port0),
             case gen_tcp:listen(Port, [Type | AddrOpts] ++ ListenOpts) of
                 {ok, Socket} ->
                     ListenAddrs = tcp_listen_addrs(Socket),
+
+                    Cache = libp2p_swarm_sup:cache(TID),
```

Comment:
Aren't most of these types of calls wrapped by a function in libp2p_swarm? eg `libp2p_swarm:cache()`

Reply:
I did like peer book `libp2p_swarm_sup:register_peerbook` and `libp2p_swarm_sup:peerbook`

Reply:
Right but there's _also_ a libp2p_swarm:peerbook function that calls the _sup function

The idea is that libp2p_swarm is the supported API of the swarm. The sup is just an implementation tactic.

Could you add that cache accessor in libp2p_swarm (including an @doc if you could) @macpie ?